### PR TITLE
Added temporary work around for finding a page type in Uno until Uno implements IXamlMetadataProvider

### PR DESCRIPTION
**Pull request by xperiandri**
@jeromelaban

### CODE REVIEW

File: src/BehaviorsSDKManaged/Microsoft.Xaml.Interactions/Core/NavigateToPageAction.cs [hunk 1]
```diff
@@ -134,13 +133,29 @@ public object Execute(object sender, object parameter)
...
+            // Temporary work around for Uno until it implements IXamlMetadataProvider
+#if __UWP__
+            var pageType = xamlType.UnderlyingType;
+#else
+            Type pageType = null;
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+            for (int i = 0; (i < assemblies.Length) && (pageType == null); i++)
+            {
+                pageType = assemblies[i].GetType(this.TargetPage, false);
+            }
```
Comment (jeromelaban) [inline src/BehaviorsSDKManaged/Microsoft.Xaml.Interactions/Core/NavigateToPageAction.cs]:
```suggestion
```suggestion
      var pageType  = AppDomain.CurrentDomain.GetAssemblies()
         .Select(a => a.GetType(this.TargetPage, false))
         .FirstOrDefault(t => t != null);
```
```

**Review by jeromelaban — COMMENTED** [review]
Thanks for the update!

Try not to change the formatting of the file. This project has no editorconfig file, so this make it a bit more difficult :)

This new search algo is valid but the original code may also be available to Uno at some point to offer better performance. 

It's probably best to change the fallback to the Assemblies search only for Uno, and keep the original code as-is.

File: src/BehaviorsSDKManaged/Microsoft.Xaml.Interactions/Core/NavigateToPageAction.cs (hunk 1 above)
Comment (xperiandri) [reply src/BehaviorsSDKManaged/Microsoft.Xaml.Interactions/Core/NavigateToPageAction.cs]: Fixed

**Comment by xperiandri** [comment]
@jeromelaban fixed according to your recommendations

**Review by jeromelaban — APPROVED** [review]

**Comment by xperiandri** [comment]
@jeromelaban what to do? I supposed that with conditional compilation `#if __UWP__` I excluded UWP
But looks like not

**Comment by jeromelaban** [comment]
Use `#if !HAS_UNO` instead

**Comment by xperiandri** [comment]
@jeromelaban done. Run again, please

**Comment by xperiandri** [comment]
@jeromelaban excuse me, please, I forgot to change the second entry. Run again

**Pull request merged**